### ISSUE TITLE
Slicing a Sparse Tensor?

### ISSUE THREAD

**Issue by yiqun-liu** [tensorflow/tensorflow#3947]
Currently, sparse tensors don't support indexing, and the tf.sparse_split() operation can't split a sparse tensor according to a given proportion. If I want to select a portion of the training set (which is a sparse tensor) as the validation set, is it possible?

**Comment by prb12** [comment]
This is a general question better suited to StackOverflow.  Could you please re-ask there.

**Comment by ksindi** [comment]
Duplicate of https://github.com/tensorflow/tensorflow/issues/1588

**Comment by soloice** [comment]
@haruna1998 I think there maybe a workaround.
You can split a sparse tensor then concatenate some parts of it.
For example:
```
indices = [[0, 0], [1,4], [2, 3], [3, 2]]
values = [1.0, 2.0, 3.0, 4.0]
num_training_samples, num_validation_samples = 3, 1
st = tf.SparseTensor(indices, values, dense_shape=(4, 5))
st_rows = tf.sparse_split(sp_input=st, num_split=4, axis=0)
training_sparse_tensor = tf.sparse_concat(axis=0, sp_inputs=st_rows[:num_training_samples])
validation_sparse_tensor = tf.sparse_concat(axis=0, sp_inputs=st_rows[-num_validation_samples:])
```
Note: not tested. But it should work.

**Comment by trdavidson** [comment]
FYI: there is now a tf.sparse_slice, see https://www.tensorflow.org/api_docs/python/tf/sparse_slice
